### PR TITLE
Keep http_plugin_impl alive while connection objects are alive

### PR DESCRIPTION
**Pull request by vzqzhang**
## Change Description

When the nodeos is killed, the thread_pool and associated io_context of the http_plugin_impl should keep alive for the lifetime of the connection object 

## Change Type
**Select ONE**
- [ ] Documentation

- [ ] Stability bug fix

- [x] Other

- [ ] Other - special case

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

**Comment by nksanthosh** [comment]
@heifner Can you please review this PR? Thanks!

**Comment by heifner** [comment]
@nksanthosh this is all my code Qing just made the PR. Needs to be reviewed by someone else.

File: plugins/http_plugin/http_plugin.cpp [hunk 1]
```diff
@@ -938,12 +951,15 @@ namespace eosio {
          my->thread_pool->stop();
       }
 
+      // release http_plugin_impl_ptr shared_ptrs captured in url handlers
+      my->url_handlers.clear();
```
Comment (swatanabe-b1) [inline plugins/http_plugin/http_plugin.cpp:955]: This isn't quite enough.  The thread pool's `io_context` may contain unexecuted handlers that hold `shared_ptrs` to the `http_plugin_impl`.

**Review by swatanabe-b1 — CHANGES_REQUESTED** [review]

File: plugins/http_plugin/http_plugin.cpp (hunk 1 above)
Comment (heifner) [reply plugins/http_plugin/http_plugin.cpp:955]: I'm not following. The `io_context` destructor will destroy all the unexecuted handlers that hold `shared_ptrs` to `http_plugin_impl`.
https://www.boost.org/doc/libs/develop/doc/html/boost_asio/reference/io_context/_io_context.html

Do you just mean the comment is not clear? Or do you see an issue here?

Comment (swatanabe-b1) [reply plugins/http_plugin/http_plugin.cpp:955]: You're only stopping the thread pool, not resetting it.  This leaves a `shared_ptr` cycle.

Comment (heifner) [reply plugins/http_plugin/http_plugin.cpp:955]: Ah. Thanks!

**Review by swatanabe-b1 — APPROVED** [review]

**Pull request merged**